src/url.rs

```diff
@@ -155,6 +155,14 @@ impl PyUrl {
         (self.__str__(),)
     }
 
+    fn __truediv__(&self, other: &str) -> PyResult<Self> {
+        self.join(other, true)
+    }
+
+    fn __floordiv__(&self, other: &str) -> PyResult<Self> {
+        self.join(other, false)
+    }
```

Comment:
Ok, sorry I missed these in the last round of review. I think the difference between the `/` and `//` operators here is subtle and hard to document.

I think better we just have `/`, and make it so that it matches the default of `append_trailing_slash=False`. This will also simplify testing, I think.

```suggestion
    fn __truediv__(&self, other: &str) -> PyResult<Self> {
        self.join(other, false)
    }
```

Reply:
Oh, I see. Yikes, there are so many subleties here!

It seems to me that our `.join()` method really works like `urllib.parse.urljoin` when it comes to semantics, e.g.

```python
>>> urllib.parse.urljoin("https://foo.com/a", "b")
'https://foo.com/b'
```

versus pathlib's

```python
>>> pathlib.Path("/foo/a").joinpath("b")
PosixPath('/foo/a/b')
```

Given these are inconsistent, I think we should perhaps back away from trying to have pathlib-like semantics at all.

Would you be open to the idea of dropping the operators from the PR completely, so we can get `.join()` merged? We could then open a `pydantic` issue to discuss the design of the operators and move forward with an implementation when there's consensus?

Reply:
Alternatively we could also have `joinpath()` which works like `Pathlib` and doesn't accept query string or fragments as the whole input?

And then could have `/` operator work like `joinpath`? 🤔

Reply:
`joinpath()` would certainly make things cleaner. Should I implement `joinpath()` in this PR, or should we drop the operators for now and discuss it in the issues instead?

Reply:
I think without comment from anyone else, let's just do `.join()` here and then follow-up with an issue in the main `pydantic` repo where we can discuss `.joinpath()` and operators.

Reply:
I think it would be great to have more time to discuss the semantics (does it need to match `urllib`? What about other libraries like [`furl`](https://github.com/gruns/furl)? Should be double check with the current RFCs? We should also check what was said in [this discussion](https://discuss.python.org/t/31109)).

Reply:
> I think it would be great to have more time to discuss the semantics (does it need to match `urllib`? What about other libraries like [`furl`](https://github.com/gruns/furl)? Should be double check with the current RFCs? We should also check what was said in [this discussion](https://discuss.python.org/t/31109)).


Really sorry for the late response. The main URL joining part is handled by [rust-url's join method](https://docs.rs/url/latest/url/struct.Url.html#method.join) which implements the [WHATWG URL spec](https://url.spec.whatwg.org/). So, the `/` operator semantics are handled by the libraries in our case, python libraries like `urllib` and `furl`. While `furl` provides `/` operator, `urllib` does not. `urllib.parse.urljoin` and `furl.furl.join` follows the [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986#section-5.2.2) to resolve the new URL. 

`furl` is using `/` operator for only adding path to URL like [pathlib.Path](https://github.com/gruns/furl/blob/da386f68b8d077086c25adfd205a4c3d502c3012/README.md?plain=1#L290).

I think the `furl` approach is good. I should not have written my function signature as `signature=(path, append_trailing_slash=false)` but rather as `signature=(url, append_trailing_slash=false)` because the argument can be a relative or absolute url, not just path.

IMO, we can have `Url.join()` for URL joining, similar to `furl.furl.join`, and `Url.__truediv__` for the sole purpose of adding a path without a trailing slash, just like `furl.furl.__truediv__`.